src/NServiceBus.Newtonsoft.Json/JsonMessageSerializer.cs

```diff
@@ -23,14 +23,23 @@ class JsonMessageSerializer : IMessageSerializer
             Func<Stream, JsonReader> readerCreator,
             Func<Stream, JsonWriter> writerCreator,
             JsonSerializerSettings settings,
-            string contentType)
+            string contentType,
+            TypeNameHandling typeNameHandling = TypeNameHandling.None)
         {
             this.messageMapper = messageMapper;
 
             if (settings == null)
             {
-                settings = new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.Auto };
-                log.Warn($"The default {nameof(JsonSerializerSettings)} for NServiceBus.Newtonsoft.Json use TypeNameHandling.Auto for backwards compatibility. This is a potential security vulnerability and it is recommended to use TypeNameHandling.None if possible. To disable this warning, provide a custom {nameof(JsonSerializerSettings)} instance to 'endpointConfiguration.UseSerialization<NewtonsoftSerializer>().Settings'. Refer to the Json.NET serializer documentation at https://docs.particular.net/ for further details.");
+                settings = new JsonSerializerSettings { TypeNameHandling = typeNameHandling };
+                if (typeNameHandling == TypeNameHandling.Auto)
+                {
+                    log.Warn($"The default {nameof(JsonSerializerSettings)} for NServiceBus.Newtonsoft.Json use TypeNameHandling.Auto for backwards compatibility. This is a potential security vulnerability and it is recommended to use TypeNameHandling.None if possible. To disable this warning, provide a custom {nameof(JsonSerializerSettings)} instance to 'endpointConfiguration.UseSerialization<NewtonsoftSerializer>().Settings'. Refer to the Json.NET serializer documentation at https://docs.particular.net/ for further details.");
```

Comment:
A quick review I was wondering how a user could even set this to Auto without passing `JsonSerializerSettings`? It seems this is only internal? If its not internal... shouldn't it be internal?

Reply:
Second, I think we could modify the url to direct to the docs search page with the component ID.

Reply:
You are correct - in fact the typeNameHandling parameter is not needed at all in V3 as the old serializer is not implemented and throws an error. 
We have removed the parameter and the warning message which no longer applies.